plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/extensions/Naming.xtend

```diff
@@ -397,72 +379,43 @@ class Naming {
 
 	def dispatch access(Method it) '''«shortName»'''
 	
-	def dispatch access(Expression it, EventDefinition event) {
-		'''«it.featureOrReference.access».iface.«event.name.asEscapedIdentifier.raised»'''
-	}
-	
-	def dispatch access(Expression it, EObject event) {
-		'''/*TODO*/'''
-	}
-	
 	def dispatch access(Enumerator it) {
 		'''«name.asEscapedIdentifier»'''
 	}
 
 	def dispatch access(OperationDefinition it) '''«asFunction»'''
 
-	def dispatch access(Event it) {
-		val container = eContainer
-		if(container instanceof ComplexType) {
-			return '''iface.«name.asIdentifier.raised»'''
-		}
-		'''«scHandle»->«scope.instance».«name.asIdentifier.raised»'''
-	}
+	def dispatch access(Event it) '''«IF needsHandle»«scHandle»«ENDIF»->«scope.instance».«name.asIdentifier.raised»'''
 
 	def dispatch access(TimeEvent it) '''«scHandle»->«scope.instance».«shortName.raised»'''
 
-	def dispatch access(EObject it) '''#error cannot access elements of type «getClass().name»'''
+	def dispatch access(EObject it) '''#error cannot access elements of type «getClass.name»'''
 
-	def valueAccess(Declaration it){
-		val container = eContainer
-		if(container instanceof ComplexType){
-			return '''iface.«name.asIdentifier.value»'''
-		}
-		'''«scHandle»->«scope.instance».«name.asIdentifier.value»'''
-	}
+	def dispatch valueAccess(Declaration it)'''«IF needsHandle»«scHandle»->«ENDIF»«scope.instance».«name.asIdentifier.value»'''
```

Comment:
What happens if `scope` is null, e.g. because the declaration comes from a header type (C domain)?